### PR TITLE
Bugfix our blocks are not sent

### PR DESCRIPTION
**Pull request by sgaragagghu**

### CODE REVIEW

File: client/pool.c [hunk 1]
```diff
@@ -1049,11 +1049,17 @@ void *pool_block_thread(void *arg)
 
 		if(b) {
 			processed = 1;
-			b->field[0].transport_header = 2;
-
-			res = xdag_add_block(b);
-			if(res > 0) {
+			int our = (uintptr_t)b & (uintptr_t)1;
```
Comment (jonano614) [inline client/pool.c]: What is the purpose of this construction: `(uintptr_t)b & (uintptr_t)1` ?

**Comment by sgaragagghu** [comment]
@jonano614 master doesn't work... i think it should work :)

File: client/pool.c [hunk 2]
```diff
@@ -1523,12 +1527,17 @@ void xdag_append_new_block(struct xdag_block *b)
...
+		b->field[0].transport_header = 0;
+	}
+
 	pthread_mutex_lock(&g_pool_mutex);
 
 	if(!g_firstb) {
 		g_firstb = g_lastb = b;
 	} else {
-		g_lastb->field[0].transport_header = (uint64_t)(uintptr_t)b;
+		((struct xdag_block*)((uintptr_t)g_lastb & ~(uintptr_t)1))->field[0].transport_header = (uint64_t)(uintptr_t)b;
```
Comment (jonano614) [inline client/pool.c]: It is not good... `(struct xdag_block*)((uintptr_t)g_lastb & ~(uintptr_t)1))`

File: client/pool.c [hunk 3]
```diff
@@ -1542,7 +1551,15 @@ struct xdag_block * xdag_first_new_block(void)
 
 	if(g_firstb) {
 		b = g_firstb;
-		g_firstb = (struct xdag_block *)(uintptr_t)b->field[0].transport_header;
+		if((uintptr_t)b & (uintptr_t)1) { // if our block
+			b = (struct xdag_block*)((uintptr_t)b & ~(uintptr_t)1); // change last bit to 0 to have correct memory address
```
Comment (jonano614) [inline client/pool.c]: Please, don't do such hacks.

**Pull request closed without merging**